### DISCUSSION TITLE
Pass a list of input parameters to a query

### DISCUSSION THREAD

**Discussion by Sandrichka22** [simolus3/drift#2934] [Q&A] [answered]
I am using the [Dart API](https://drift.simonbinder.eu/docs/dart-api/) and I am defining my tables and queries in SQL.
I want to create a query that would be passing some input parameters but I am not sure if that is possible.
So the query I am looking to create would be something like this, to check if what I am passing (list of ids and types) is already in DB:

checkForNewPosts: SELECT COUNT(*)
    FROM PostsEntity P
    WHERE id IN ('id1', 'id2', 'id3', ...) AND type IN ('type1', 'type2', 'type3', ...);
    
I would appreciate if anybody has some feedback on this. Thanks in advance.

**Comment by Mike278** [comment, marked as answer]
Have a look at [this part](https://drift.simonbinder.eu/docs/sql-api/drift_files/#arrays) of the docs. 

In your case it'd look something like:

```
checkForNewPosts: SELECT COUNT(*)
FROM PostsEntity P
WHERE id IN ? AND type IN ?;
```

**Reply by simolus3** [reply]
I'll mark this as answer since it's the cleanest solution if you're already using SQL for queries.

If you want to do it in Dart, you can use [isIn](https://drift.simonbinder.eu/api/drift/expression/isin): `id.isIn(['id1', 'id2', 'id3', ...]) & type.isIn(['type1', 'type2', 'type3', ...])`
